tardis/visualization/tools/sdec_plot.py

```diff
@@ -424,6 +439,95 @@ def from_hdf(cls, hdf_fpath):
             )
         )
 
+    def _parse_species_list(self, species_list):
+        """
+        Parse user requested species list and create list of species ids to be used.
+
+        Parameters
+        ----------
+        species_list : list of species to plot
+            List of species (e.g. Si II, Ca II, etc.) that the user wants to show as unique colours.
+            Species can be given as an ion (e.g. Si II), an element (e.g. Si), a range of ions
+            (e.g. Si I - V), or any combination of these (e.g. species_list = [Si II, Fe I-V, Ca])
+
+        """
+
+        if species_list is not None:
+            # check if there are any digits in the species list. If there are, then exit.
+            # species_list should only contain species in the Roman numeral
+            # format, e.g. Si II, and each ion must contain a space
+            if any(char.isdigit() for char in " ".join(species_list)) == True:
+                raise ValueError(
+                    "All species must be in Roman numeral form, e.g. Si II"
+                )
+            else:
+                full_species_list = []
+                for species in species_list:
+                    # check if a hyphen is present. If it is, then it indicates a
+                    # range of ions. Add each ion in that range to the list as a new entry
+                    if "-" in species:
+                        # split the string on spaces. First thing in the list is then the element
+                        element = species.split(" ")[0]
+                        # Next thing is the ion range
+                        # convert the requested ions into numerals
+                        first_ion_numeral = roman_to_int(
+                            species.split(" ")[-1].split("-")[0]
+                        )
+                        second_ion_numeral = roman_to_int(
+                            species.split(" ")[-1].split("-")[-1]
+                        )
+                        # add each ion between the two requested into the species list
+                        for ii in np.arange(
+                            first_ion_numeral, second_ion_numeral + 1
+                        ):
+                            full_species_list.append(
+                                element + " " + int_to_roman(ii)
+                            )
```

Comment:
James already got here first and suggested putting them all to ii so it can be searched

Reply:
So, I might be wrong as I don't know what the code style specifically is for TARDIS, but I thought the general rule was to have counters labelled e.g. aa, bb, ii, jj etc. so that they can be easily searched for in the code?

Reply:
But perhaps we can discuss in meeting tomorrow

Reply:
> I thought the general rule was to have counters labelled e.g. aa, bb, ii, jj etc. so that they can be easily searched for in the code?
I'm not aware of it - also, I don't think searching should be problem since our code editors are advanced enough to match single character variables

Reply:
Also, I didn't mention it earlier but a good rule of readability is to give meaningful variable name - even to counters. For e.g. here it could be `ion_num`

Reply:
I've changed them all to meaningful names